miden/tests/integration/operations/decorators/asmop.rs

```diff
@@ -0,0 +1,414 @@
+use crate::build_debug_test;
+use processor::AsmOpInfo;
+use vm_core::{Felt, Operation};
+
+#[test]
+fn asmop_one_span_block_test() {
+    let source = "begin push.1 push.2 add end";
+    let test = build_debug_test!(source);
+    let vm_state_iterator = test.execute_iter();
+    let expected_vm_state = vec![
+        VmStatePartial {
+            clk: 0,
+            asmop: None,
+            op: None,
+        },
+        VmStatePartial {
+            clk: 1,
+            asmop: None,
+            op: Some(Operation::Span),
+        },
+        VmStatePartial {
+            clk: 2,
+            asmop: Some(AsmOpInfo::new("push.1".to_string(), 2, 1)),
+            op: Some(Operation::Pad),
+        },
+        VmStatePartial {
+            clk: 3,
+            asmop: Some(AsmOpInfo::new("push.1".to_string(), 2, 2)),
+            op: Some(Operation::Incr),
+        },
+        VmStatePartial {
+            clk: 4,
+            asmop: Some(AsmOpInfo::new("push.2".to_string(), 1, 1)),
+            op: Some(Operation::Push(Felt::new(2))),
+        },
+        VmStatePartial {
+            clk: 5,
+            asmop: Some(AsmOpInfo::new("add".to_string(), 1, 1)),
+            op: Some(Operation::Add),
+        },
+        VmStatePartial {
+            clk: 6,
+            asmop: None,
+            op: Some(Operation::End),
+        },
+    ];
+    let mut vm_state = Vec::new();
+    for state in vm_state_iterator {
+        vm_state.push(VmStatePartial {
+            clk: state.as_ref().unwrap().clk,
+            asmop: state.as_ref().unwrap().asmop.clone(),
+            op: state.as_ref().unwrap().op,
+        });
+    }
```

Comment:
This pattern repeats several times - maybe create a helper function for this?

Reply:
Thanks. Done.